File: frontend/src/types/directive.ts
import type { Classification } from './common'

export type DirectiveStatus = 'nhap' | 'da_ban_hanh'

export const DIRECTIVE_STATUS_LABELS: Record<DirectiveStatus, string> = {
  nhap: 'Bản nháp',
  da_ban_hanh: 'Đã ban hành',
}

export interface Directive {
  id: number
  title: string
  content: string
  status: DirectiveStatus
  classification: Classification
  author_id: number
  author_full_name: string
  created_at: string
  recipient_count: number
  acknowledged_count: number
  acknowledged_by_me: boolean
}

export interface DirectiveCreate {
  title: string
  content: string
  status: DirectiveStatus
  classification: Classification
}

export interface DirectiveAckUser {
  user_id: number
  full_name: string
  role: string
  acknowledged_at: string | null
}

export interface DirectiveAckReport {
  directive_id: number
  recipient_count: number
  acknowledged_count: number
  acknowledged: DirectiveAckUser[]
  pending: DirectiveAckUser[]
}

File: backend/app/schemas/directive.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.common import Classification

DirectiveStatus = Literal["nhap", "da_ban_hanh"]


class DirectiveCreate(BaseModel):
    title: str = Field(..., max_length=255)
    content: str
    status: DirectiveStatus = "nhap"
    classification: Classification = "noi_bo"


class DirectiveOut(BaseModel):
    id: int
    title: str
    content: str
    status: DirectiveStatus
    classification: Classification
    author_id: int
    author_full_name: str
    created_at: datetime
    # thong ke muc do quan triet
    recipient_count: int
    acknowledged_count: int
    acknowledged_by_me: bool

    model_config = ConfigDict(from_attributes=True)


class DirectiveAckUser(BaseModel):
    user_id: int
    full_name: str
    role: int
    acknowledged_at: datetime | None = None


class DirectiveAckReport(BaseModel):
    directive_id: int
    recipient_count: int
    acknowledged_count: int
    acknowledged: list[DirectiveAckUser]
    pending: list[DirectiveAckUser]
